cli_tools/gce_vm_image_import/cli/args.go

```diff
@@ -26,147 +26,52 @@ import (
-	dataDiskFlag       = "data_disk"
-	osFlag             = "os"
-	customWorkflowFlag = "custom_translate_workflow"
-)
-
-// ImportArguments holds the structured results of parsing CLI arguments,
-// and optionally allows for validating and populating the arguments.
-type ImportArguments struct {
-	ExecutionID           string
-	ClientID              string
-	ClientVersion         string
-	CloudLogsDisabled     bool
-	ComputeEndpoint       string
-	ComputeServiceAccount string
-	WorkflowDir           string
-	CustomWorkflow        string
-	DataDisk              bool
-	Description           string
-	Family                string
-	GcsLogsDisabled       bool
-	ImageName             string
-	Labels                map[string]string
-	Network               string
-	NoExternalIP          bool
-	NoGuestEnvironment    bool
-	Oauth                 string
-	BYOL                  bool
-	OS                    string
-	Project               string
-	Region                string
-	ScratchBucketGcsPath  string
-	Source                Source
-	SourceFile            string
-	SourceImage           string
-	StdoutLogsDisabled    bool
-	StorageLocation       string
-	Subnet                string
-	SysprepWindows        bool
-	Started               time.Time
-	Timeout               time.Duration
-	UefiCompatible        bool
-	Zone                  string
+// imageImportArgs holds the arguments passed by the user, and facilitates creating
+// importer.ImageImportRequest.
+type imageImportArgs struct {
+	ClientID      string
+	ClientVersion string
+	Region        string
+	SourceFile    string
+	SourceImage   string
+	Started       time.Time
+	importer.ImageImportRequest
 }
 
-// NewImportArguments parses args to create an ImportArguments instance.
-// No validation occurs; to validate, use ValidateAndPopulate.
-func NewImportArguments(args []string, workflowDir string) (ImportArguments, error) {
+func parseArgsFromUser(argsFromUser []string) (imageImportArgs, error) {
 	flagSet := flag.NewFlagSet("image-import", flag.ContinueOnError)
 	// Don't write parse errors to stdout, instead propagate them via an
 	// exception since we use flag.ContinueOnError.
 	flagSet.SetOutput(ioutil.Discard)
-
-	parsed := ImportArguments{
-		WorkflowDir: workflowDir,
-		Started:     time.Now(),
+	parsed := imageImportArgs{
+		Started: time.Now(),
+		ImageImportRequest: importer.ImageImportRequest{
+			ExecutionID: path.RandString(5),
+		},
 	}
-
 	parsed.registerFlags(flagSet)
-
-	return parsed, flagSet.Parse(args)
+	return parsed, flagSet.Parse(argsFromUser)
 }
 
-// ValidateAndPopulate parses, validates, and populates the arguments.
-func (args *ImportArguments) ValidateAndPopulate(populator param.Populator,
-	sourceFactory SourceFactory) (err error) {
+func (args *imageImportArgs) populate(populator param.Populator,
+	sourceFactory importer.SourceFactory) (err error) {
+
+	if args.ClientID == "" {
+		return fmt.Errorf("%s has to be specified", importer.ClientFlag)
+	}
```

Comment:
This is validation. Now that the method is called just `populate`, doesn't seem it belongs here anymore.

Reply:
Good call, renamed to `populateAndValidate`, and added a a comment explaining that a minimal amount of validation occurs here, and the bulk of validation occurs in the importer module.